core/src/crypto.rs

```diff
@@ -25,8 +25,12 @@ impl Secret {
         self.secret.diffie_hellman(pub_key)
     }
 
-    pub fn gen_cipher(&mut self) -> Aes256 {
+    pub fn gen_pub_cipher(&mut self) -> Aes256 {
```

Comment:
Also no need for `mut` here, as far as I can tell.

Reply:
gen_pub_cipher() needs a mut self, but the other doesn't no :)